src/lib/output/renderer.ts

```diff
@@ -201,7 +201,24 @@ export class Renderer extends ChildableComponent<Application, RendererComponent>
                 }
             }
 
-            this.theme = this.addComponent('theme', new DefaultTheme(this, path));
+            let filename = Path.join(path, 'theme.js');
+            if (!FS.existsSync(filename)) {
+                this.theme = this.addComponent('theme', new DefaultTheme(this, path));
+            } else {
+                try {
+                    const plugin = require(filename);
+                    if(typeof plugin === 'function') {
+                        plugin(this.application);
```

Comment:
Trouble is that breaks existing functionality.  There was no `plugin(this.application)` before this PR for theme.js and existing theme.js (that I'm aware of) are sub-classes of DefaultTheme and expect to be passed (Renderer, BasePath).  See [typedoc-markdown-theme](https://github.com/kimamula/typedoc-markdown-theme/blob/master/src/theme.ts).

The consistency I mentioned was a new alternative for a theme.js file which would be consistent with typedoc plugins but maintain backward compatibility for existing theme.js usage.